rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -104,6 +104,25 @@ create_node(
 
   try {
     node_impl = new CustomParticipantInfo();
+
+    node_impl->leave_middleware_default_qos = false;
+    const char * env_var = "RMW_FASTRTPS_USE_QOS_FROM_XML";
+    // Check if the configuration from XML has been enabled from
+    // the RMW_FASTRTPS_USE_QOS_FROM_XML env variable.
+    char * config_env_val = nullptr;
+#ifndef _WIN32
```

Comment:
Since we are already depending on `rcutils`, https://github.com/ros2/rcutils/blob/master/include/rcutils/get_env.h#L62 would be a reasonable choice here.

Reply:
I checked this function before implementing the feature and I chose to implement the getenv again in this package because the function in rcutils is not safe. It stores the result of the "getenv" in a static variable that could be overwritten by another call before taking the result for this feature.

Reply:
Sounds good. Thanks.